### PR TITLE
Added codable models for Assignment

### PR DESCRIPTION
**Pull request by agavgavi**
Created Codable models for User, Token, Member, and Team. For now, the JSON decoder is stored in the ViewController but will be moved to manager classes. (It is only there to make sure that we have the decoder's date set to the proper format).

### CODE REVIEW

**Review by wtpisaac — CHANGES_REQUESTED** [review]
I believe the JSON decoding logic can be stored within the classes, and these can be bound to an optional initializer. I will attempt to make these changes.

File: Sluggo/Models/User.swift [hunk 1]
```diff
@@ -14,4 +14,42 @@ struct UserRecord: Codable {
...
+        guard let jsonData = fromJSONString.data(using: .utf8) else { return nil }
+        
+        // Attempt decoding
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .iso8601
+        
+        guard let userRecord = try? decoder.decode(UserRecord.self, from: jsonData) else { return nil }
+        
+        // Initialize using standard initializer
+        self.init(id: userRecord.id, email: userRecord.email, first_name: userRecord.first_name, last_name: userRecord.last_name, username: userRecord.username)
```
Comment (tdimhcsleumas) [inline Sluggo/Models/User.swift:43]: what's userRecord's type? You could probably abstract most of the json decoding to a base class and then define the initializer to build from a json object

File: Sluggo/Models/Member.swift [hunk 2]
```diff
@@ -18,5 +18,42 @@ struct MemberRecord: Codable {
...
+
+        // Attempt decoding
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .iso8601
+
+        guard let memberRecord = try? decoder.decode(MemberRecord.self, from: jsonData) else {
+            print("Failed to decode JSON data into object representation for user object initialization.")
+            return nil
+        }
+          return memberRecord
```
Comment (tdimhcsleumas) [inline Sluggo/Models/Member.swift:37]: Good work! :ship:

**Review by tdimhcsleumas — APPROVED** [review]

**Comment by tdimhcsleumas** [comment]
Good work! :ship:

**Pull request merged**